### ISSUE TITLE
[net6] Using LLVM for `ios-arm` fails

### ISSUE THREAD

**Issue by spouliot** [xamarin/xamarin-macios#12546]
### Steps to Reproduce

1.  Checkout `release/6.0.1xx-rc.1` (have not tried on `main` but it's likely the same)

2. Apply this diff

```diff
diff --git a/tests/dotnet/Makefile b/tests/dotnet/Makefile
index cb6f96212..61aee264d 100644
--- a/tests/dotnet/Makefile
+++ b/tests/dotnet/Makefile
@@ -50,10 +50,10 @@ build-oldnet:
        $(SYSTEM_MSBUILD) $(XBUILD_VERBOSITY) size-comparison/MySingleView/oldnet/MySingleView.csproj $(COMMON_ARGS) /bl:$@.binlog $(MSBUILD_VERBOSITY)
 
 build-dotnet: $(TARGETS)
-       $(DOTNET6) build size-comparison/MySingleView/dotnet/MySingleView.csproj --runtime ios-arm64 /p:MtouchUseLlvm=true $(COMMON_ARGS) /bl:$@.binlog $(MSBUILD_VERBOSITY)
+       $(DOTNET6) build size-comparison/MySingleView/dotnet/MySingleView.csproj /p:MtouchUseLlvm=true $(COMMON_ARGS) /bl:$@.binlog $(MSBUILD_VERBOSITY)
 
 run-dotnet: $(TARGETS)
-       $(DOTNET6) build -t:Run size-comparison/MySingleView/dotnet/MySingleView.csproj --runtime ios-arm64 /p:MtouchUseLlvm=true $(COMMON_ARGS)
+       $(DOTNET6) build -t:Run size-comparison/MySingleView/dotnet/MySingleView.csproj /p:MtouchUseLlvm=true $(COMMON_ARGS)
 
 # this will break the signature, so app won't run anymore. Use it only to compare final size w/legacy
 # https://github.com/xamarin/xamarin-macios/issues/11445
diff --git a/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj b/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj
index 03153d664..5f8c25d75 100644
--- a/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj
+++ b/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>net6.0-ios</TargetFramework>
-    <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
+    <RuntimeIdentifier>ios-arm</RuntimeIdentifier>
     <OutputType>Exe</OutputType>
   </PropertyGroup>
   <ItemGroup>
```

3. `cd tests/dotnet/ then `make build-dotnet`

### Expected Behavior

Successful build

### Actual Behavior

```
make build-dotnet
cp -c ../../global6.json global.json
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/dotnet build size-comparison/MySingleView/dotnet/MySingleView.csproj /p:MtouchUseLlvm=true /p:Platform=iPhone /p:Configuration=Release /bl:build-dotnet.binlog /nologo /verbosity:quiet
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/sdk/6.0.100-rc.1.21422.7/MSBuild.dll /nologo -consoleloggerparameters:Summary -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/sdk/6.0.100-rc.1.21422.7/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/sdk/6.0.100-rc.1.21422.7/dotnet.dll -maxcpucount -property:MtouchUseLlvm=true -property:Platform=iPhone -property:Configuration=Release -restore -verbosity:m /verbosity:quiet /bl:build-dotnet.binlog size-comparison/MySingleView/dotnet/MySingleView.csproj
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile System.Runtime.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile MySingleView.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile Xamarin.iOS.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile System.Private.CoreLib.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]

Build FAILED.

/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile System.Runtime.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile MySingleView.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile Xamarin.iOS.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
/Users/poupou/git/net6/rc1/xamarin-macios/builds/downloads/dotnet-sdk-6.0.100-rc.1.21422.7-osx-x64/packs/Microsoft.iOS.Sdk/15.0.100-rc.1.516/targets/Xamarin.Shared.Sdk.targets(805,3): error : Failed to AOT compile System.Private.CoreLib.dll, the AOT compiler exited with code 1 [/Users/poupou/git/net6/rc1/xamarin-macios/tests/dotnet/size-comparison/MySingleView/dotnet/MySingleView.csproj]
    0 Warning(s)
    4 Error(s)

Time Elapsed 00:00:26.53

```

### Environment

This was tested on the `2f30a4676954de83da23c27cef102456f505903d` of branch: `release/6.0.1xx-rc.1`

### Build Logs

[build-dotnet.binlog.zip](https://github.com/xamarin/xamarin-macios/files/7047512/build-dotnet.binlog.zip)

### Example Project (If Possible)

In repo

**Comment by spouliot** [comment]
This was found trying a _fat_ app - and that make those fails since LLVM is a global [1] setting.

[1] maybe it should not be ? there's nothing that should stop only one of the slice to be built with LLVM while the other is not.

**Comment by rolfbjarne** [comment]
This is filed here: https://github.com/dotnet/runtime/issues/58112

**Comment by filipnavara** [comment]
This was fixed on the dotnet/runtime side, correct LLVM option will be used after next runtime bump.

**Comment by filipnavara** [comment]
Should be fixed now.

**Comment by rolfbjarne** [comment]
Confirmed; this is fixed now.

**Comment by spouliot** [comment]
It's also fixed inside the `release/6.0.1xx-rc.1` so we can build release _fat_ apps with LLVM
https://gist.github.com/spouliot/e487514f4376d44823e81829fc44a521
